app/src/main/java/org/android/go/sopt/util/extension/ViewExt.kt

```diff
@@ -0,0 +1,15 @@
+package org.android.go.sopt.util.extension
+
+import android.view.View
+import com.google.android.material.snackbar.Snackbar
+import org.android.go.sopt.util.OnSingleClickListener
+
+/** Register a callback to be invoked when this view is clicked. If this view was clicked within 1 second, the callback will not be invoked. */
+fun View.setOnSingleClickListener(onSingleClick: (View) -> Unit) {
+    setOnClickListener(OnSingleClickListener { onSingleClick(it) })
+}
```

Comment:
```suggestion
inline fun View.setOnSingleClickListener(crossinline onSingleClick: (View) -> Unit) {
    setOnClickListener(OnSingleClickListener { onSingleClick(it) })
}
```
이렇게도 적을 수 있을 것 같아. 

관련 키워드: [인라인 함수](https://sabarada.tistory.com/176), [crossinline](https://leveloper.tistory.com/171)

Reply:
앞으로 고차함수를 사용할 때에는 inline, reified, noinline, crossinline 키워드를 통해 불필요한 객체 생성을 줄여봐야겠군요 😲
2주차 과제하면서 한번 활용해보겠습니다! 덕분에 많이 배우고 갑니다 ㅎㅎ